### PR TITLE
Fix Sphinx warnings in re library docs

### PR DESCRIPTION
**Pull request by wulmer**
As addition to PR #106995 this PR fixes the remaining Sphinx warnings.

There are two issues:
1. Missing `re.U` definition
2. Reference on C function `scanf`

For 2., I chose to use exclamation mark to disable the reference link, but I am not sure if `scanf` should be included into #106948.

----
:books: Documentation preview :books:: https://cpython-previews--107044.org.readthedocs.build/

### CODE REVIEW

**Review by methane — APPROVED** [review]

**Comment by methane** [comment]
LGTM. But I wait for another review from better English writer before merge.

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by AlexWaygood** [comment]
Ah, I see the conflict for the 3.11 backport is (at least partly) due to the fact that the `.nitignore` file doesn't exist on the 3.11 branch. That's somewhat annoying...

**Comment by AlexWaygood** [comment]
@wulmer, are you able to take care of the 3.11 backport?

**Comment by wulmer** [comment]
> @wulmer, are you able to take care of the 3.11 backport?

@AlexWaygood : I think so. I have read the comments above about the `cherry_picker` command, do I have to use it? Or would it be ok to just commit to the 3.11 branch?

**Comment by AlexWaygood** [comment]
> > @wulmer, are you able to take care of the 3.11 backport?
> 
> @AlexWaygood : I think so. I have read the comments above about the `cherry_picker` command, do I have to use it? Or would it be ok to just commit to the 3.11 branch?

Either is absolutely fine :)

**Pull request merged**
